pkg/client/node.go

```diff
@@ -361,7 +358,16 @@ func patchServerSpec(node *k3d.Node) error {
 	node.Labels[k3d.LabelServerAPIHost] = node.ServerOpts.KubeAPI.Host
 	node.Labels[k3d.LabelServerAPIPort] = node.ServerOpts.KubeAPI.Binding.HostPort
 
-	node.Args = append(node.Args, "--tls-san", node.ServerOpts.KubeAPI.Host) // add TLS SAN for non default host name
+	// If the runtime is docker, attempt to use the docker host
+	if runtime.ID() == "docker" {
+		dockerHost := runtime.GetHost()
+		if dockerHost != "" {
+			node.Labels[k3d.LabelServerAPIHostIP] = runtime.GetHost()
+			node.Labels[k3d.LabelServerAPIHost] = runtime.GetHost()
+		}
+	}
```

Comment:
Picking that up again (not sure if GitHub is fooling me here or not :thinking: 
```suggestion
	if runtime == runtimes.Docker {
		dockerHost := runtime.GetHost()
		if dockerHost != "" {
			node.Labels[k3d.LabelServerAPIHostIP] = dockerHost
			node.Labels[k3d.LabelServerAPIHost] = dockerHost
		}
	}
```